src/cheri_insts.sail

```diff
@@ -44,20 +44,35 @@ union clause ast = CGetOffset : (regidx, regidx)
 union clause ast = CGetAddr   : (regidx, regidx)
 union clause ast = CGetFlags  : (regidx, regidx)
 
+/*!
+ * The least significant [cap_hperms_width] bits of integer register *rd* are
+ * set equal to the **perms** field of capability register *cs1*; bits
+ * [cap_uperms_shift] to [cap_uperms_shift]+[cap_uperms_width]-1 of *rd* are set
+ * equal to the **uperms** field of *cs1*.
+ * The other bits of \emph{rd} are set to zero.
+ */
 function clause execute (CGetPerm(rd, cs1)) =
 {
   let capVal = C(cs1);
   X(rd) = EXTZ(getCapPerms(capVal));
   RETIRE_SUCCESS
 }
 
+/*!
+ * Integer register *rd* is set equal to the zero-extended **flags** field of
+ * capability register *cs1*.
+ */
 function clause execute (CGetFlags(rd, cs1)) =
 {
   let capVal = C(cs1);
   X(rd) = EXTZ(getCapFlags(capVal));
   RETIRE_SUCCESS
 }
 
+/*!
+ * Integer register *rd* is set equal to the sign-extended **otype** field
+ * of capability register *cs1*.
```

Comment:
That's not quite true; normal otypes are positive, reserved otypes are negative, and the split is asymmetric.

Reply:
And arguably like bounds the otype is architecturally an XLEN-bit value with restricted values due to architectural compression...

Reply:
MIPS has this:
```\subsection*{Notes}

\begin{itemize}
\item
If the capability is unsealed, a value of -1 is returned.
For hardware-interpreted \cotype{}s (see \cref{tab:archotypes}) a sign-extended
(negative) value is returned; for software-defined \cotype{}s
(see \cref{sec:model-sealedcapabilities}) \insnmipsref{CGetType} returns
a zero-extended value.
\end{itemize}
```

Reply:
I could just drop the sign-extended and people can look at the sail to figure out the value?

Reply:
Yeah that might be easiest, though we could add something reminding people that software-defined otypes are positive and architecturally-reserved otypes are negative.

Reply:
Uh, under Notes, that is.